core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/impl/api/joinWith.kt

```diff
@@ -25,7 +25,7 @@ internal class JoinedDataRowImpl<A, B>(
     override val right: DataRow<B> = DataRowImpl(index1, rightOwner)
 }
 
-internal fun <A, B> DataFrame<A>.joinWithImpl(
+internal inline fun <A, B> DataFrame<A>.joinWithImpl(
```

Comment:
hmm I'm doubting the benefits outweigh the downsides for inlining a function as large as this. Remember inlining copies over everthing in this function to the execution site increasing code size and compile time for the user. @nikita, what do you think?

Reply:
I agree, let's keep it a regular fun